metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -205,218 +195,613 @@ def get_workunits_internal(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit
                     profile_requests, profiler, platform=self.platform
                 )
 
-            oauth_schema = "Entities"
-            if sql_config.include_oauth:
-                yield from self.loop_oauth(inspector, oauth_schema, sql_config)
-
     def get_database_properties(
-        self, inspector: Inspector, database: str
+        self, inspector: VerticaInspector, database: str
     ) -> Optional[Dict[str, str]]:
         try:
-            custom_properties = inspector._get_database_properties(database)  # type: ignore
+            custom_properties = inspector._get_database_properties(database)
             return custom_properties
+
         except Exception as ex:
             self.report.report_failure(
                 f"{database}", f"unable to get extra_properties : {ex}"
             )
         return None
 
     def get_schema_properties(
-        self, inspector: Inspector, database: str, schema: str
+        self, inspector: VerticaInspector, database: str, schema: str
     ) -> Optional[Dict[str, str]]:
         try:
-            custom_properties = inspector._get_schema_properties(schema)  # type: ignore
+            custom_properties = inspector._get_schema_properties(schema)
             return custom_properties
         except Exception as ex:
             self.report.report_failure(
                 f"{database}.{schema}", f"unable to get extra_properties : {ex}"
             )
         return None
 
-    def _process_table(
+    def loop_tables(  # noqa: C901
```

Comment:
The changes should only be in `process_table` - there should be no changes in `loop_tables`

Schema-level queries should happen inside of the vertica inspector and then get cached.

Table-level queries should refresh the cache by calling the schema level query once, and then reading the cache for every subsequent call.